src/main-oe/sgxlkl_run_oe.c

```diff
@@ -2032,25 +2050,35 @@ int main(int argc, char* argv[], char* envp[])
         pthread_setname_np(sgxlkl_threads[i], "ENCLAVE");
     }
 
-    long exit_status = 0;
-
-    for (int i = 0; i < econf->ethreads; i++)
-    {
-        pthread_join(sgxlkl_threads[i], &return_value);
-        if (i == 0)
-        {
-            exit_status = (long)return_value;
-        }
-    }
-    sgxlkl_host_verbose_raw("\n");
-
-    if (oe_enclave)
+    // Wait for the terminating ethread to exit the enclave
+    pthread_mutex_lock(&terminating_ethread_exited_mtx);
+    ret = pthread_cond_wait(
+        &terminating_ethread_exited_cv, &terminating_ethread_exited_mtx);
+    if (ret != 0)
+        sgxlkl_host_fail("Failed to wait for enclave to finish: ret=%i\n", ret);
+    long exit_status = (long)enclave_return_status;
```

Comment:
Why an explicit cast from `long` to `long`?

Reply:
Fixed.